File: passenger-service/src/test/java/io/simakkoi9/passengerservice/config/PostgresTestContainer.java
package io.simakkoi9.passengerservice.config;

import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.testcontainers.containers.PostgreSQLContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;
import org.testcontainers.utility.DockerImageName;

@Testcontainers
public class PostgresTestContainer {
    private static final String IMAGE_VERSION = "postgres:16";
    private static final String DATABASE_NAME = "testdb";
    private static final String USERNAME = "testuser";
    private static final String PASSWORD = "testpassword";

    @Container
    @ServiceConnection
    private static PostgreSQLContainer<?> postgreSQLContainer =
            new PostgreSQLContainer<>(DockerImageName.parse(IMAGE_VERSION))
                    .withDatabaseName(DATABASE_NAME)
                    .withUsername(USERNAME)
                    .withPassword(PASSWORD);

    static {
        postgreSQLContainer.start();
    }
}

File: driver-service/src/test/kotlin/io/simakkoi9/driverservice/config/PostgresTestContainer.kt
package io.simakkoi9.driverservice.config

import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.testcontainers.containers.PostgreSQLContainer
import org.testcontainers.junit.jupiter.Container
import org.testcontainers.junit.jupiter.Testcontainers

@Testcontainers
object PostgresTestContainer {

    private const val IMAGE_VERSION = "postgres:16"
    private const val DATABASE_NAME = "testdb"
    private const val USERNAME = "test"
    private const val PASSWORD = "test"

    @Container
    @ServiceConnection
    val POSTGRES_CONTAINER = PostgreSQLContainer(IMAGE_VERSION)
        .withDatabaseName(DATABASE_NAME)
        .withUsername(USERNAME)
        .withPassword(PASSWORD)

} 